make/clang-tidy

```diff
@@ -0,0 +1,16 @@
+
+tidy_base_checks=*perf*,*modernize*,*misc*,*google*,*readability*,*optin*
```

Comment:
Can we add a link to a clang-tidy doc here? That would help someone trying to understand these flags. If I am getting this right this does all the checks in `*perf*,*modernize*,*misc*,*google*,*readability* and *optin*` except the ones that are excluded?

I was exploring these a few months ago, but did not get near as far as you did now :) This is really cool!

Reply:
Yep added a link to the top of the file!